zerver/lib/bugdown/__init__.py

```diff
@@ -889,6 +948,12 @@ def run(self, root: Element) -> None:
                 else:
                     add_embed(root, url, extracted_data)
 
+            quoted_message = self.message_quote(url)
+            if quoted_message is not None:
```

Comment:
This can be simplified to `if is_message_url(url):` if you rename the function above. If not, still it seems like you aren't using the return value stored in `quoted_message` elsewhere.